backend/api/student/simulation_instances.py

```
@@ -888,11 +887,30 @@ async def complete_simulation_instance(
+            user_progress.last_activity = now
+            # Best-effort: mark current scene as completed if not already
+            try:
+                sp = db.query(SceneProgress).filter(
+                    SceneProgress.user_progress_id == user_progress.id,
+                    SceneProgress.scene_id == user_progress.current_scene_id
+                ).first()
+                if sp and sp.status != "completed":
+                    sp.status = "completed"
+                    sp.completed_at = now
+            except Exception:
+                pass
```

Ruff (0.14.2)

`try`-`except`-`pass` detected, consider logging the exception

(S110)